Cart View 수정 / Main Page View 작성

joey-younggap opened the pull request:
[Add] Cart View 수정 / Main Page View 작성
- cart 추가하기, 보여주기 최종 테스트 후 완성본으로 업데이트
- main 화면에 보여질 products list view 작성

[Fix] Signup View 수정
- signup 시 존재하는 email의 경우 if문으로 예외처리로 변경

[Fix] 기타 상기 변동 사항 외 urls 등 변경

* REST url 법칙에 따른 url 변경은 오늘 발표 후 진행할 예정입니다.

Commit:
[Add] Cart View 수정 / Main Page View 작성
- cart 추가하기, 보여주기 최종 테스트 후 완성본으로 업데이트
- main 화면에 보여질 products list view 작성

[Fix] Signup View 수정
- signup 시 존재하는 email의 경우 if문으로 예외처리로 변경

[Fix] 기타 상기 변동 사항 외 urls 등 변경

devops-andrew commented on customer/views.py:
```diff
@@ -4,17 +4,17 @@
 
         try:
+            if Customer.objects.filter(email=data['email']).exists():
+                return JsonResponse({'message':'Already registered email'}, status=400)
```
이 메세지는 프론트에서 활용하나요? 메세지코드 컨벤션이 제각각입니다. 확인해주세요.
```suggestion
                return JsonResponse({'message':'EMAIL_ALREADY_EXIST'}, status=400)
```

joey-younggap replied:
수정 사항 반영하였습니다.

devops-andrew commented on customer/views.py:
```diff
@@ -28,8 +28,6 @@ def post(self, request):
             return JsonResponse({'message':'INVALID INPUT'}, status=400)
```
TypeError를 만나기보다 만나기 전에 미리 검사하는게 더 좋을것 같네요.

joey-younggap replied:
수정 사항 반영하였습니다.

devops-andrew commented on freeporkbelly_main/urls.py:
```diff
@@ -3,6 +3,6 @@
     path('product', include('product.urls')),
-    #path('order', include('order.urls')),
+    path('order', include('order.urls')),
     #path('support', include('support.urls'))
```
사용안하는 코드는 제거해주세요.

joey-younggap replied:
수정 사항 반영하였습니다.

Commit:
wip_cart_delete

devops-andrew commented on product/views.py:
```diff
@@ -9,6 +9,22 @@
+
+class MainView(View):
+    def get(self, request):
+        products = Product.objects.all()[0:6]    # 프로덕트 중에서 6개만 뽑음
```
리미트 값을 준다면 하드코딩이 아니라 
PRODUCT_MAIN_LIMIT = 6
과 같이 상수로 지정하고 처리하는게 가독성에 좋습니다.
```suggestion
        products = Product.objects.all()[:PRODUCT_MAIN_LIMIT]    # 프로덕트 중에서 6개만 뽑음
```

joey-younggap replied:
수정 사항 반영하였습니다.

Commit:
master_merge_main_view

Commit:
[Fix] Cart View 수정
- 카트 처음으로 생성할 때 카트를 새롭게 생성한 후 카트 아이템 담도록 로직 수정
(처음 카트 만드는 유저일 경우, 에러나는 부분 해결)
- delete 함수 코드 오타 수정 (테스트 완료)

[Fix] 피드백 사항 수정
- SignUp 오류 메시지 다른 부분과 통일화
- TypeError 관련 부분 사전 점검 및 해당 코드삭제
- usrls.py에 주석처리된 부분 삭제
- 리미트 값 부여시 PRODUCT_MAIN_LIMIT = 6 적용

joey-younggap commented:
피드백 사항 반영하여, 수정하여 다시 푸시 날렸습니다.
아울러 피드백 사항 외에 Cart View에서 
- post : 유저가 맨 처음 카트아이템을 추가할 때 카트를 신규로 생성하는 로직 추가
  (맨 처음 카트아이템을 추가할 때 발생했던 오류 해결)
- delete : 오타 수정 (테스트 완료)

soheon-lee commented on order/views.py:
```diff
@@ -9,16 +9,19 @@
-            cart = Cart.objects.filter(customer_id=request.customer.id).last()
-            cart_items = CartItem.objects.filter(cart_id=cart.id, prodcut_id=data['product_id'])
+            if Cart.objects.filter(customer_id=data['customer_id']).exists():       # 해당 ID의 카트가 존재하면, 그거 가져옴
+                cart = Cart.objects.filter(customer_id=data['customer_id']).last()
```
.get 메소드를 사용하지 않고, filter().last()를 하시는 이유가 있을까요?

soheon-lee commented on order/views.py:
```diff
@@ -9,16 +9,19 @@
-            cart = Cart.objects.filter(customer_id=request.customer.id).last()
-            cart_items = CartItem.objects.filter(cart_id=cart.id, prodcut_id=data['product_id'])
+            if Cart.objects.filter(customer_id=data['customer_id']).exists():       # 해당 ID의 카트가 존재하면, 그거 가져옴
+                cart = Cart.objects.filter(customer_id=data['customer_id']).last()
```
get() 메소드를 사용하지 않고, filter().last()를 사용하시는 이유 설명 부탁드립니다.

joey-younggap replied:
이 부분은 아까 전에 말씀 나눠서 설명 드렸습니다.

soheon-lee commented on order/views.py:
```diff
@@ -42,27 +45,27 @@ def post(self, request):
-                'sub_img_url' : Proudct.objects.get(id=cart.product_id).sub_img_url,
-                'selected_option' : OptionItems.objects.get(id=cart_item.option_id).name,
-                'sales_price' : Proudct.objects.get(id=cart.product_id).sales_price,
+                'name' : Product.objects.get(id=cart_item.product_id).name,
+                'sub_img_url' : Product.objects.get(id=cart_item.product_id).sub_img_url,
+                'selected_option' : OptionItems.objects.get(id=cart_item.selected_option_id).name,
+                'sales_price' : Product.objects.get(id=cart_item.product_id).sales_price,
                 'quantity' : cart_item.quantity
```
cart_item에서 바로 name에 접근 하실 수 있습니다. 로직 수정부탁드립니다.

joey-younggap replied:
오~ 감사합니다. 방금 전 로직 수정했습니다.

soheon-lee commented on product/views.py:
```diff
@@ -9,6 +9,24 @@
+class MainView(View):
+    def get(self, request):
+        
+        PRODUCT_MAIN_LIMIT = 6                   # 메인페이지에 6개 상품만 보여줌
```
숫자 6은 프론트엔드에서 쿼리스트링으로 받아주세요

joey-younggap replied:
프론트엔드에서 쿼리스트링으로 받아오도록 로직 수정했습니다.
다만, 없을 경우 기본값을 6으로 하였습니다.

Commit:
[Fix] Product/Order View 피드백 사항 반영
- 메인페이지 상품보여주는 갯수 프론트엔드에서 받는 것으로 설정
- cart_item으로 바로 프로덕트  데이터 접근하도록 수정

soheon-lee commented:
LGTM
